### PR TITLE
Dynamic rescan progress loading

### LINKED ISSUE #386: Include status/progress bars to blockchain and UTXO rescans
Currently, toggling from screen to screen is the only way to know the real-time progress of the two rescans.  Noobs love progress bars - we demand moar!  :p

### PR DESCRIPTION
**Pull request by ben-kaufman**
Fix #386

### CODE REVIEW

File: src/cryptoadvance/specter/templates/includes/sidebar/sidebar.jinja [hunk 1]
```diff
@@ -53,8 +53,46 @@
...
+                        if (walletRescanData || walletRescanPercents) {
+                            location.reload();
+                        }
+                    }
+                } catch(e) {
+                    console.log('Caught error:', e);
+                    return { success: false, error: e };
+                }
+            }
+            let fetchUTXORescanInterval = setInterval(fetchUTXORescanProgress, 2000);
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/templates/includes/sidebar/sidebar.jinja:94]: Same as for wallet rescan. `setTimeout` is better.

**Comment by ben-kaufman** [comment]
Fixed both, should be good now even with multiple wallets rescans at the same time.

**Comment by stepansnigirev** [comment]
When I stay in the settings of the wallet after clicking "rescan" button, when rescan ends the page is refreshed, which triggers rescan again. I think it refreshes the page that effectively makes the same post request over and over.
So if I leave Specter open on the settings page of the wallet I get inifinte rescans...
The same for both UTXO and simple wallet rescan.

File: src/cryptoadvance/specter/templates/includes/sidebar/sidebar.jinja [hunk 2]
```diff
@@ -53,8 +53,49 @@
...
+                            walletRescanPercentsSettings.innerHTML = (Math.round(parseFloat(utxoRescanProgress) * 100) / 100).toFixed(2);
+                        }
+                        setTimeout(fetchUTXORescanProgress, 1000);
+                    } else {
+                        document.getElementById('utxo_rescan_progress').style.display = 'none';
+                        if (walletRescanData) {
+                            walletRescanData.style.display = 'none';
+                        }
+                        if (walletRescanData || walletRescanPercentsSettings) {
+                            location.reload();
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/templates/includes/sidebar/sidebar.jinja:89]: avoid POSTing the same request again:

```suggestion
                            window.location = window.location.href;
```

File: src/cryptoadvance/specter/templates/includes/sidebar/components/sidebar_wallet_list_item.jinja [hunk 3]
```diff
@@ -12,13 +12,62 @@
...
+                            }
+                            setTimeout(fetchWalletRescanProgress, 1000);
+                        } else {
+                            document.getElementById('{{ wallet.alias }}_rescan_progress').style.display = 'none';
+                            document.getElementById('{{ wallet.alias }}_sidebar_rescan_progress_container').style.display = 'none';
+                            if (walletRescanData) {
+                                walletRescanData.style.display = 'none';
+                            }
+                            if (walletRescanData || walletRescanPercents || walletRescanPercentsSettings) {
+                                location.reload();
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/templates/includes/sidebar/components/sidebar_wallet_list_item.jinja:61]: Avoid POSTing the same request again
```suggestion
                                window.location = window.location.href;
```

**Pull request merged**